data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -424,16 +426,25 @@ private void processRecord(final AcknowledgementSet acknowledgementSet, final Re
                 bufferAccumulator.add(record);
                 break;
             } catch (Exception e) {
+                if (!paused) {
```

Comment:
I would suggest to pause only when we are approaching `max.poll.ms` limit, which we can also increase to 10 mins or more. I have seen this exception many times in scale testing, but buffer usually flushes after few retries and poll timeout does not expire. Pause/Resume may have some performance impact, like I read it flushes queued messages and stops fetching further records from broker, which is really not necessary if it's a momentary blip.

Reply:
Sure, we can pause only after few retries of getting the needed buffer and that should help in doing all this in a momentary blip case.